### ISSUE TITLE
address tight coupling with ksp & kotlin version

### ISSUE THREAD

**Issue by jillesvangurp** [jwstegemann/fritz2#636]
**Is your feature request related to a problem? Please describe.**

I just tried to update to Kotlin 1.6.20, which was just released. It failed because the ksp version has to match the kotlin version used in the lenses jar.

**Describe the solution you'd like**

Would it be possible to relax this so patch releases of Kotlin are not a problem? Alternatively, can you release a new version against 1.6.20?

**Describe alternatives you've considered**

I tried manually adding the ksp dependency with the later version but that doesn't seem to work.

**Additional context**

**Comment by chausknecht** [comment]
If the KSP version of our plugin has to match the Kotlin version of the application I cannot imagine a decoupled solution. Although it feels quite odd to me, that there is such a strong coupling to be honest.

I do not know, if someone from our team has already tried to use Kotlin 1.6.20 with fritz2 already.

But I am quite sure the fritz2-1.0-RC1 release is quite close and that will be based upon the current Kotlon version, if we do not encounter any problems.

I can only encourage you, to build fritz2 locally on your own with the adapted versions! If that works, why not create a PR? It is really not that hard :-)

**Comment by jillesvangurp** [comment]
Yeah that's what I suspected. I don't really have a good development environment for Fritz2. I should fix that probably but it will take me some time. I can upgrade the dependencies of course and create a PR. 

Looking forward to the release candidate btw.

**Comment by jillesvangurp** [comment]
@chausknecht I created a PR that does the upgrade to 1.6.21. Sorry this took so long.

I would recommend using the refreshVersions plugin. That's a bit bigger refactor of the build but it's a great plugin for staying on top of versions of dependencies.

I can do that as well if you need me to. 

Probably a lot of the npm dependencies need updating as well.
